src/code/dialogs/checklist.js

```diff
@@ -51,12 +54,24 @@ const OperationChecklistDialog = WDialog.extend({
   _displayDialog: async function () {
     const operation = getSelectedOperation();
     loadFaked(operation);
+
+    // defaults to sorting by op order
+    if (localStorage[this.SORTBY_KEY] == null) {
+      localStorage[this.SORTBY_KEY] = 0;
+    }
+    if (localStorage[this.SORTASC_KEY] == null) {
+      localStorage[this.SORTASC_KEY] = "true";
+    }
+
     this.sortable = this.getListDialogContent(
       operation,
       operation.links.concat(operation.markers),
-      0,
-      false
-    ); // defaults to sorting by op order
+      localStorage[this.SORTBY_KEY],
+      localStorage[this.SORTASC_KEY] == "true"
+    );
+
+    this.sortable.sortByStoreKey = this.SORTBY_KEY;
+    this.sortable.sortAscStoreKey = this.SORTASC_KEY;
```

Comment:
shouldn't this be in getListDialogContent instead ?

Reply:
It just needs to be set once per dialog instance, and before the user can interact with it. Setting these in getListDialogContent would just keep setting it to the same values, every time it is updated for example. The fields are only used when the user clicks on a column header.

Reply:
`getListDialogContent` creates a new sortable, so on update, the localStorage is not updated on sorting

Reply:
True, that is why it is set after getListDialogContent. However, getListDialogContent is used by 2 other dialogs (markerList and linkList) they each need to set their own store key and not use the store key for checklist.

Reply:
The sorting is not saved because of this, for instance
 - open checklist
 - reverse a link
 - sort on a column
 - close checklist
 - open checklist
Here, last sort is not restored

`this.SORTBY_KEY` should have the value you set in the child dialog for marker/link, so it's not an issue

Reply:
Yes, this will fix the issue you found. And the value of this.SORTKEY is indeed the value of the linklist and markerlist keys in checklist. Really makes the code hard to read, but anyway, it works!